src/ConfigFileParser.cpp

```diff
@@ -0,0 +1,534 @@
+}
+
+/**
+ * @brief Reads and checks the value of the directive in the current line of the config file
+ *
+ * @details This function is called when the directive is valid.
+ *          It calls the appropriate function to read the value of the directive.
+ *          It throws an exception if the value is invalid.
+
+ * @param directive Is the the directive which value is being read and checked
+ */
+void ConfigFileParser::readServerDirectiveValue(const std::string& directive, const std::string& value)
+{
+    if (directive == "listen") {
+        readIpAddress(value);
+        readPort(value);
+    } else if (directive == "root")
+        readRootPath(SERVER, value);
+}
+
+/**
+ * @brief Gets the directive from the line
+ *
+ * @param line String containing the directive and value
+ * @return std::string The extracted directive
+ */
+std::string ConfigFileParser::getDirective(const std::string& line) const
+{
+    size_t firstWhiteSpaceIndex = line.find_first_of(" \t\n\v\f\r");
+    std::string directive = line.substr(0, firstWhiteSpaceIndex);
+
+    removeLeadingAndTrailingSpaces(directive);
+
+    return directive;
+}
+
+/**
+ * @brief Gets the value from the line
+ *
+ * @param line String containing the directive and value;
+ * @return std::string  The extracted value
+ */
+std::string ConfigFileParser::getValue(const std::string& line) const
+{
+    if (getDirective(line) == "location")
+        return line;
+
+    size_t semicolonIndex = line.find(';');
+
+    size_t firstWhiteSpaceIndex = line.find_first_of(" \t\n\v\f\r");
+    std::string value = line.substr(firstWhiteSpaceIndex, semicolonIndex - firstWhiteSpaceIndex);
+
+    removeLeadingAndTrailingSpaces(value);
+
+    return value;
+}
+
+/**
+ * @brief Processes the line after the directive and value got extracted
+ *
+ * It can be the case that multiple directorives are in the same line.
+ * Therefore the line must still get read until all directives got processed.
+ *
+ * If the line still contains more then one directive, the processed directive gets skipped.
+ * If the line does not contain any more directives, the line gets cleared.
+ *
+ * @param line The current line
+ */
+void ConfigFileParser::processRemainingLine(std::string& line) const
+{
+    size_t semicolonIndex = line.find(';');
+
+    if (line.size() > semicolonIndex + 1) {
+        line = line.substr(semicolonIndex + 1);
+        removeLeadingAndTrailingSpaces(line);
+    } else
+        line.clear();
+}
+
+/**
+ * @brief Reads the current line of the server config and does several checks
+ *
+ * How a line gets processed:
+ * 1. Gets checked if it contains a semicolon.
+ * 2. The directive of of the resulting string gets extracted and checked.
+ * 3. The value of the directive gets extracted and checked.
+ *
+ * If the line contains more then one directive, this process gets repeated for each directive.
```

Comment:
Spelling: double "of" and "more than one" at the end

Reply:
Done